activerecord/lib/active_record/core.rb

```diff
@@ -19,11 +19,23 @@ module Core
       # retrieved on both a class and instance level by calling +logger+.
       class_attribute :logger, instance_writer: false
 
-      ##
-      # :singleton-method:
-      #
-      # Specifies the job used to destroy associations in the background
-      class_attribute :destroy_association_async_job, instance_writer: false, instance_predicate: false, default: false
+      class_attribute :_destroy_association_async_job, instance_writer: false, instance_predicate: false, default: "ActiveRecord::DestroyAssociationAsyncJob"
+
+      # The job class used to destroy associations in the background.
+      def self.destroy_association_async_job
+        if _destroy_association_async_job.is_a?(String)
+          self._destroy_association_async_job = _destroy_association_async_job.constantize
+        end
+        _destroy_association_async_job
+      rescue NameError => error
+        if error.name == _destroy_association_async_job
+          raise NameError, "Missing or invalid class or ancestor for destroy_association_async_job: #{error.name.inspect}"
```

Comment:
I don't think we can dynamically determine the missing ancestor from just the string.

I'm thinking it is simpler to just capture all the name-errors and slightly rewrite them to be `Unable to load destroy_association_async_job: #{error.message}`. In the case of ActiveJob not being loaded that would look like:

```
Unable to load destroy_association_async_job: uninitialized constant ActiveJob::Base
```

Reply:
> I don't think we can dynamically determine the missing ancestor from just the string.

Agreed.  What I meant was the `if error.name == _destroy_association_async_job` only handles the case where `_destroy_association_async_job` itself references an invalid class (e.g. typo, not defined, etc).  The other branch (that was there) would re-raise any other `NameError`, such as `ActiveJob::Base` being undefined.  So the custom error message did not need to mention "ancestor" (or so I was inquiring).

Reply:
Addendum: It looks like `NameError#name` returns a symbol rather than a string.  So my original code was wrong — it should have been `if error.name.to_s == _destroy_association_async_job`.

Reply:
Oh! That makes sense now. I hope you're ok with the more generic overriding all NameErrors messages.

Reply:
I think it provides enough information to figure out the problem. :+1: